File: 6. Python/EX45.py
# Inicializa contadores
idAux = 0
qtdPar = 0
flag = False

# Solicita a quantidade de números
qtd = int(input("# Quantos numeros voce vai digitar: "))

# Cria vetores para os números e os pares
aux = [0 for i in range(qtd)]
numVet = [0 for i in range(qtd)]

# Coleta os números
for i in range(qtd):
    numVet[i] = int(input("# Digite um numero: "))

    # Verifica se é par
    if numVet[i] % 2 == 0:
        aux[idAux] = numVet[i]  # Armazena par
        idAux += 1              # Incrementa índice
        qtdPar += 1             # Conta pares
        flag = True             # Indica que há pares

# Exibe os pares, se existirem
if flag:
    print("\n# NUMEROS PARES:")
    for i in range(idAux):
        print(f" {aux[i]}", end="")
    print(f"\n\n# QUANTIDADE DE PARES = {qtdPar}")

File: 4. Java/EX45.java
package Apostila;

import java.util.Scanner;

public class EX45
{
    public static void main(String[] args)
    {
        // Inicializa variáveis
        int idaAux = 0, qtdPar = 0, qtd; // Contadores para pares e total
        
        // Lê a quantidade de números
        try (Scanner ent = new Scanner(System.in))
        {
            // Lê a quantidade de números
            System.out.print("# Quantos numeros voce vai digitar: ");
            qtd = ent.nextInt();
            
            // Declara vetores para números e pares
            int [] numVet = new int[qtd];
            int [] aux =  new int[qtd];
            
            // Lê os números e verifica se são pares
            for (int i = 0; i < qtd; i++)
            {
                System.out.print("# Digite um numero: ");
                numVet[i] = ent.nextInt();
                
                // Verifica se o número é par
                if (numVet[i] % 2 == 0)
                {
                    aux[idaAux] = numVet[i]; // Armazena par no vetor auxiliar
                    idaAux++; // Incrementa o índice
                    qtdPar++; // Conta os pares
                }
            }
            
            // Exibe números pares, se houver
            if (qtdPar > 0)
            {
                System.out.print("\n# NUMEROS PARES:");                
                // Lista os números pares
                for (int i = 0; i < qtdPar; i++)
                {
                    System.out.print(" " + aux[i]);
                }
            }

            // Exibe quantidade de pares
            System.out.println("\n\n# QUANTIDADE DE PARES = " + qtdPar);
        }
    }
}
